kythe/go/platform/kzip/info/info.go

```diff
@@ -43,10 +46,30 @@ func KzipInfo(f kzip.File, scanOpts ...kzip.ScanOption) (*apb.KzipInfo, error) {
 
 	err := kzip.Scan(f, func(rd *kzip.Reader, u *kzip.Unit) error {
 		kzipInfo.TotalUnits++
-		corpusInfo(u.Proto.GetVName().GetCorpus()).Units[u.Proto.GetVName().GetLanguage()]++
+
+		// The unit VName generally does not specify a corpus, so we have to
+		// determine corpus from the source files.
```

Comment:
If it does we should defer to that, no?

Reply:
It's probably more useful to detect that all source files share a corpus and use that for validation/informational purposes.

Reply:
Sure, but given the documented semantics of this we should actually use those semantics when determining the source file corpus, rather than just ignoring the CompilationUnit corpus entirely.

Reply:
I don't think we're ignoring a CompilationUnit corpus.  It's empty, and therefore, not set in proto3-land.  In this case, all of the source files have the same corpus, however, and the CompilationUnit corpus really should have that same corpus (or at the very least set a differing corpus).  I think we're just covering for bad extraction here.

We could add a flag for this behavior.  Call it "--inference" or something, but I'm not sure where would want the previous behavior of attributing the compilation unit to no corpus.

Reply:
I simplified this a bit and put all mentions of the corpus (from unit vname and source vnames) into a stringset. At the end, if there's one value in the set, we have a consistent vname. Otherwise a warning is logged. This way there's no priority/precedence between unit and source file vnames, but if specified in both places, we do check that they're consistent

Reply:
@justbuchanan that works, thanks!

> I don't think we're ignoring a CompilationUnit corpus. It's empty, and therefore, not set in proto3-land. In this case, all of the source files have the same corpus, however, and the CompilationUnit corpus really should have that same corpus (or at the very least set a differing corpus). I think we're just covering for bad extraction here.
> 
> We could add a flag for this behavior. Call it "--inference" or something, but I'm not sure where would want the previous behavior of attributing the compilation unit to no corpus.

The documented algorithm for determining a file VName corpus is:
 `unit.required_input[path].v_name.corpus OR unit.v_name.corpus`

The CompilationUnit VName corpus is not *necessarily* empty and, if present, should be used as a fallback corpus for file VNames.  It's also somewhat relevant to the discussion we had the other day about "build corpus" vs "attribute corpus" as the CU corpus is the former and file corpus the latter.